[weeweh] iP

weeweh opened the pull request:
### Meet ###
## BanterBot ##

>"Banter is often a proof of want of intelligence." - Jean de la Bruyere [(source)](https://www.azquotes.com/quotes/topics/banter.html)

BanterBot frees your mind of having to remember things you need to do. It is:
+ text-based 
+ easy to learn
+ ~~HARD~~ FUN and FAST to use
+ humorous..? 😂

Steps to get BanterBot:
1. download it from [here](https://github.com/weeweh/ip/releases).
2. add your tasks.
3. let it manage your tasks for you 😉
4. have fun!!

Guess what: it is **FREE**!

Features include:
- [x] Create and Delete Tasks (Todos, Deadlines, and Events)
- [x] Search for Tasks
- [x] Find what Deadlines are due
- [ ] Find when Events are happening (Coming Soon..?)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
``` java
public class Main {
    public static void main(String[] args) {
        new BanterBot("data/tasks.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

weeweh committed:
Gave chatbot a new name
Implemented skeletal version

weeweh committed:
Skeletal version echos commands entered, exits when user types the command bye.

weeweh committed:
Added ability to store text entered and display back when requested.

weeweh committed:
Add the ability to mark tasks as done and change the status back to not done.

weeweh committed:
Add support for tracking three types of tasks: todos, deadline, event

weeweh committed:
Test using the I/O redirection technique

weeweh committed:
no message

weeweh committed:
Used Enums instead for command

weeweh committed:
added new exceptions

weeweh committed:
Revert "added new exceptions"

This reverts commit bcf298d6791a52c74c1f4fbc38792175b9a4b8c8.

weeweh committed:
added exception handling

weeweh committed:
added delete functionality

weeweh committed:
add reading and writing of tasks.txt

weeweh committed:
Merge branch 'branch-Level-7'

weeweh committed:
add date parsing in Deadline
add due command

weeweh committed:
refactor code to be more OOP

weeweh committed:
Merge branch 'branch-A-MoreOOP'

weeweh committed:
Organise classes into packages

weeweh committed:
Organise classes into packages

weeweh committed:
Merge branch 'branch-A-Packages'

weeweh committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

weeweh committed:
Merge branch 'branch-A-Gradle'

* branch-A-Gradle:
  Bump gradle and lib version
  Add Gradle support

weeweh committed:
Add JUnit Tests

weeweh committed:
Merge branch 'branch-A-JUnit'

weeweh committed:
Add Jar Fille for v1.0

weeweh committed:
Add JavaDoc to all classes and methods

weeweh committed:
Refactor code to coding standard

weeweh committed:
Merge branch 'branch-A-JavaDoc'

weeweh committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/BanterBot.java
#	src/main/java/helper/Parser.java
#	src/main/java/helper/Storage.java

weeweh committed:
Add more JavaDoc comments

weeweh committed:
Add Find command

weeweh committed:
Merge branch 'branch-Level-9'

weeweh committed:
Add checkstyle to gradle
Fix checkstyle errors

weeweh committed:
Merge branch 'branch-A-CheckStyle'

Elijah5399 commented on src/main/java/BanterBot.java:
```diff
@@ -0,0 +1,50 @@
+            try {
+                String fullCommand = ui.readCommand();
+                Command c = Parser.parse(fullCommand);
+                c.execute(tasks, ui, storage);
```
I really like the way you structured your entire program. You used ui.readCommand to form the fullCommand string, then formed the Command using Parser.parse, and finally executed the command using command.execute, providing tasks, ui and storage as arguments. Would you like to consider giving c a more appropriate name such as "command"?

Elijah5399 commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,47 @@
+     */
+    public boolean isExit() {
+        return false;
+    }
```
I like how you named this boolean function isExit to sound like a boolean! :+1:

Elijah5399 commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,36 @@
+ */
+public class DeleteCommand extends Command {
+    /** Command the user starts with to activate the DeleteCommand. */
+    public static final String COMMAND_WORD = "delete";
```
I like how you named this constant in all caps! :+1:

Elijah5399 commented on src/main/java/helper/Ui.java:
```diff
@@ -0,0 +1,50 @@
+public class Ui {
+    /**
+     * A line to divide responses. */
+    public static final String DIVIDER = "____________________________________________________________";
```
I noticed that you consistently use JavaDoc for methods and fields. Well done! However, do you think it would be neater to make this javadoc comment a single line instead of two lines?

Elijah5399 commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,42 @@
+     */
+    @Override
+    public String toFileString() {
+        int bool = this.isDone ? 1 : 0;
```
The naming of this variable `int bool` may be a bit misleading. Would you like to rename it to something like `int doneNumeric` instead?

weeweh committed:
Add GUI support

weeweh committed:
Merge branch 'branch-Level-10'

weeweh committed:
Add asserts to Commands and TaskList

Commands and Tasklist have made some assumptions in order for the method to function properly
(e.g. TaskList.add(index i) requires i to be more than 0 and less than the list.size()).

Adding assertions can document important assumptions that should hold for this methods to work as intended.

Let's add assertions to various points in the code to document valid assumptions that are implicit and make it as clear as possible for other developers.

weeweh committed:
Refactor complicated code to coding standard throughout codebase

Code was not written up to coding standard which could be hard to code trace and hide unwanted bugs within code.

The refactoring methods employed were:
* Some methods were extracted to smaller methods
* SLAP was attempted as much as possible
* if-else blocks were written to follow the happy path

Let's try to follow the coding standard when writing to code to ensure readability.

weeweh committed:
Merge pull request #1 from weeweh/branch-A-CodeQuality

Refactor complicated code to coding standard throughout codebase

weeweh committed:
Merge branch 'master' of https://github.com/weeweh/ip

* 'master' of https://github.com/weeweh/ip:
  Refactor complicated code to coding standard throughout codebase

weeweh committed:
Merge pull request #2 from weeweh/branch-A-Assertions

Add asserts to Commands and TaskList

weeweh committed:
Merge branch 'master' of https://github.com/weeweh/ip

* 'master' of https://github.com/weeweh/ip:
  Add asserts to Commands and TaskList

weeweh committed:
Add a new command remind that lists deadline that are due 3 days from the current time

weeweh committed:
Merge branch 'branch-B-Reminders'

* branch-B-Reminders:
  Add a new command remind that lists deadline that are due 3 days from the current time

weeweh committed:
Correct CheckStyle errors
Add 2 more unit test to Parser

weeweh committed:
Merge branch 'branch-A-MoreTesting'

weeweh committed:
Add Ui.png to docs

weeweh committed:
Add README.md to include feature list and details of each command

weeweh committed:
Merge branch 'branch-A-UserGuide'

weeweh committed:
Create new package called banterbot
Build new JAR file

weeweh committed:
Edit minor errors in README.md

weeweh committed:
Add header view

weeweh committed:
Edit website title

LuahJunYang reviewed (commented):
Nice use of OOP for the commands, I didn't consider this approach at all. Good job naming all the methods as well as they are all quite clear
